File: src/main/java/dark/leech/text/models/Repository.java
package dark.leech.text.models;

import java.util.List;

import com.google.gson.annotations.SerializedName;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class Repository {
    @SerializedName("metadata")
    private MetaData metaData;

    @SerializedName("data")
    private List<Plugin> plugins;

    @Data
    @Builder
    public static class MetaData {
        public String author;
        public String description;
    }

    @Data
    @Builder
    public static class Plugin {
        public String name;
        public String uuid;
        public String author;
        public String path;
        public double version;
        public String source;
        public String icon;
        public String description;
        public String type;
        public String locale;
        public String tag;
    }
}

File: engine/src/commonMain/kotlin/dev/haipham22/leechtext/models/Repository.kt
package dev.haipham22.leechtext.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/** Index JSON của một repo plugin (port từ models/Repository.java). */
@Serializable
data class Repository(
    @SerialName("metadata") var metaData: MetaData? = null,
    @SerialName("data") var plugins: List<Plugin>? = null,
) {
    @Serializable
    data class MetaData(
        var author: String? = null,
        var description: String? = null,
    )

    @Serializable
    data class Plugin(
        var name: String? = null,
        var uuid: String? = null,
        var author: String? = null,
        var path: String? = null,
        var version: Double = 0.0,
        var source: String? = null,
        var icon: String? = null,
        var description: String? = null,
        var type: String? = null,
        var locale: String? = null,
        var tag: String? = null,
    )
}
